lib/make-badge.js

```diff
@@ -91,64 +87,85 @@ function compressSvg(string, callback) {
 const cssColor = /^#([0-9a-fA-F]{6}|[0-9a-fA-F]{3})$/;
 
 // Inject the measurer to avoid placing any persistent state in this module.
-function makeBadge (measurer, data) {
-  if (data.format !== 'json') {
-    data.format = 'svg';
+function makeBadge (measurer, {
+  format,
+  template,
+  text,
+  colorscheme,
+  colorA,
+  colorB,
+  logo,
+  logoWidth,
+  links = ['', ''],
+}) {
+  // String coercion.
+  text = text.map(value => '' + value);
+
+  if (format !== 'json') {
+    format = 'svg';
   }
-  if (!(data.template + '-' + data.format in templates)) {
-    data.template = 'flat';
+
+  if (!(`${template}-${format}` in templates)) {
+    template = 'flat';
   }
-  if (data.template === 'for-the-badge'){
-    data.text[0] = ('' + data.text[0]).toUpperCase();
-    data.text[1] = ('' + data.text[1]).toUpperCase();
+  if (template === 'social') {
+    text[0] = capitalize(text[0]);
+  } else if (template === 'for-the-badge') {
+    text = text.map(value => value.toUpperCase());
```

Comment:
Not able to check it right now, but will this still handle numbers okay? #1280

Reply:
Yea, the coercion was happening in two places before. I moved one of them to the very top of the function.

Reply:
Ah, missed that!
That should help in avoiding any future problems too.